### PR TITLE
Add BUILD rules and separate target for fuchsia benchmarks

### PR DESCRIPTION
**Pull request by jonahwilliams**
BUILD.gn rules allow the flutter_gallery app to be built in tree for a flutter driver test. A slightly modified driver script for the performance test which works but does not currently record any performance targets.

This needs to be under a separate package to prevent the fuchsia rules from trying to run all of the tests.

Additionally, tweaks the video_demo to avoid initializing connectivity on fuchsia devices.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
I'm not sure If I can commit code which depends on packages that pub doesn't know about...

File: dev/bots/analyze.dart [hunk 1]
```diff
@@ -266,6 +266,7 @@ Future<void> _verifyNoTestPackageImports(String workingDirectory) async {
         if (path.split(file.path).contains('test_driver') ||
             name.startsWith('dev/missing_dependency_tests/') ||
             name.startsWith('dev/automated_tests/') ||
+            name.startsWith('dev/fuchsia_benchmarks') ||
```
Comment (tvolkert) [inline dev/bots/analyze.dart]: Nit: trailing slash

File: examples/flutter_gallery/lib/demo/video_demo.dart [hunk 2]
```diff
@@ -309,6 +310,9 @@ class _ConnectivityOverlayState extends State<ConnectivityOverlay> {
   @override
   void initState() {
     super.initState();
+    if (defaultTargetPlatform == TargetPlatform.fuchsia) {
+      return;
```
Comment (tvolkert) [inline examples/flutter_gallery/lib/demo/video_demo.dart]: Add a comment explaining why

**Review by tvolkert — COMMENTED** [review]
We should ask ourselves: what would this look like in the world where the Flutter & Dart Fuchsia rules live in a Flutter repo?

My initial thoughts on that, in no particular order:
* `is_fuchsia_host` goes away, because we no longer have to be building the OS to use this.
* We'll probably be relying on the following things being installed on the system to be able to build a Fuchsia Flutter package (discoverable and invokable from flutter_tools):
  * Prebuilt Fuchsia SDK tools
  * Prebuilt flutter_runner tools downloaded into `bin/cache`
* Both `package:fuchsia_remote_debug_protocol` and `package:fuchsia_driver` go away, and `flutter_tools` and `flutter_driver` both know how to Just Work in a Fuchsia context.

In that world, a lot of this code looks less Fuchsia-specific.

Given the discussion we had yesterday around Fuchsia SDK, I wonder if we should just move toward that world first so we don't start accumulating too much technical debt.

**Pull request closed without merging**